tpchgen-cli/src/plan.rs

```diff
@@ -223,212 +191,507 @@ impl Display for GenerationPlan {
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+/// output size of a table
+#[derive(Debug)]
+struct OutputSize {
+    /// Average row size in bytes
+    avg_row_size_bytes: i64,
+    /// Number of rows in the table
+    row_count: i64,
+    /// output target chunk size in bytes
+    target_chunk_size_bytes: i64,
+    /// maximum part count, if any
+    max_part_count: Option<i64>,
+}
 
-    #[test]
-    fn sf1_nation() {
-        Test::new()
-            .with_table(Table::Nation)
-            .with_format(OutputFormat::Tbl)
-            .with_scale_factor(1.0)
-            .assert(1, 1..=1)
-    }
+impl OutputSize {
+    pub fn new(table: &Table, scale_factor: f64, format: OutputFormat) -> Self {
+        let row_count = Self::row_count_for_table(table, scale_factor);
+
+        // The average row size in bytes for each table in the TPC-H schema
+        // this was determined by sampling the data
+        let avg_row_size_bytes = match format {
+            OutputFormat::Tbl | OutputFormat::Csv => match table {
+                Table::Nation => 88,
+                Table::Region => 77,
+                Table::Part => 115,
+                Table::Supplier => 140,
+                Table::Partsupp => 148,
+                Table::Customer => 160,
+                Table::Orders => 114,
+                Table::Lineitem => 128,
+            },
+            // Average row size in bytes for each table at scale factor 1.0
+            // computed using datafusion-cli:
+            // ```shell
+            // datafusion-cli -c "datafusion-cli -c "select row_group_id, count(*), min(row_group_bytes)::float/min(row_group_num_rows)::float as bytes_per_row from parquet_metadata('lineitem.parquet') GROUP BY 1 ORDER BY 1""
+            // ```
+            OutputFormat::Parquet => match table {
+                Table::Nation => 117,
+                Table::Region => 151,
+                Table::Part => 70,
+                Table::Supplier => 164,
+                Table::Partsupp => 140,
+                Table::Customer => 168,
+                Table::Orders => 75,
+                Table::Lineitem => 64,
+            },
+        };
 
-    #[test]
```

Comment:
github is rendering this diff strangely -- I renamed these tests but they are all still present along with many new tests

In my opinion, viewing whitespace blind diff makes the change much easier to understand:
- https://github.com/clflushopt/tpchgen-rs/pull/168/files?w=1